core/modules/query/TableRef.cc

```diff
@@ -205,9 +205,14 @@ std::string TableRef::sqlFragment() const {
 
 void TableRef::putTemplate(QueryTemplate& qt) const {
     auto aliasMode = qt.getTableAliasMode();
+    bool quoteIdentifiers = qt.quoteIdentifiers();
     if (QueryTemplate::USE == aliasMode) {
         if (hasAlias()) {
-            qt.append("`" + _alias + "`");
+            if (quoteIdentifiers) {
+                qt.append("`" + _alias + "`");
```

Comment:
Since this construct is seen in multiple places:
```
"`" + _alias + "`"
```
Then you may replace it as an inline function:
```
string sqlQuotedId(string const& id) const {
    return "`" + id + "`";
}
```
And use it where needed. This approach has the following benefits:
* it clearly expresses the semantic of the operation (unlike just concatenating strings)
* it prevents possible mistakes in the code
* it allows replacing MySQL-specific back ticks with the SQL standard double quote symbol `"`